collect_app/src/main/java/org/odk/collect/android/widgets/items/SelectOneWidget.java

```diff
@@ -100,15 +99,21 @@ public void onItemClicked() {
             listener.advance();
         }
 
+        onClearNextLevelsOfCascadingSelect();
         widgetValueChanged();
     }
 
+    @Override
+    public void clearAnswer() {
+        onClearNextLevelsOfCascadingSelect();
+        super.clearAnswer();
+    }
+
     /**
      * It's needed only for external choices. Everything works well and
      * out of the box when we use internal choices instead
      */
-    @Override
-    public void onClearNextLevelsOfCascadingSelect() {
+    private void onClearNextLevelsOfCascadingSelect() {
```

Comment:
I'm afraid it must be in this widget because the first level of cascading select is always `SelectOneWidget` and then next levels are represented by `ItemsetWidgets`. So if I change it like that it won't work.

Reply:
I see. I guess that's what the provided examples do show though it's certainly not a requirement. This is all kind of crazy so let's at least improve the comment and naming. How about `clearFollowingItemsetWidgets` and something like "If there are "fast external itemset" selects right after this select, assume that they are linked to the current question and clear them."

Reply:
Sounds good, will do.